pyproject.toml

```diff
@@ -17,15 +17,23 @@ classifiers = [
 ]
 requires-python = ">=3.7"
 dependencies = [
-    "podcastparser>=0.6.0",
-    "mygpoclient>=1.7",
     "dbus-python;platform_system=='Linux'",
+    "mygpoclient==1.10",
+    "podcastparser==0.6.10",
     "PyGObject",
+    "requests[socks]==2.32.3",
+    "urllib3==2.2.2",
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
+full = [
+    "eyed3",
```

Comment:
> eyed3 is optional and pulls in a lot of dependencies, so disable by default

I assume yt-dlp must be specified to use it, could the same be done for eyed3?

Reply:
> I assume yt-dlp must be specified to use it, could the same be done for eyed3?

Probably, but TBH I'm not familiar enough with the Python build system to say how useful these build variants are.

The flatpak build manifest does `make install` which currently just calls `python -m build --wheel --no-isolation` without any other arguments. Could other installers (windows, mac) use the variants somehow?

Reply:
I also know nothing about python's build system, and have never used pip.

According to a quick search, `pip install '.[yt-dlp,eyed3]'` would install gpodder along with those optional deps, after running `make build` of course. I don't know if any linux distros make use of pip in their build system, but users could theoretically install gpodder this way. There is also `pip-gui`, and it or another frontend might extract optional deps and make them more discoverable.

The Windows and Mac builds manually install each dependency.